xdsl/builder.py

```diff
@@ -55,22 +55,34 @@ class Builder(BuilderListener):
     """Operations will be inserted at this location."""
 
     def insert(self, op: OperationInvT) -> OperationInvT:
+        """
+        Inserts op at the current location and returns it.
+        """
+        self.insert_op(op)
+        return op
+
+    def insert_op(
+        self,
+        op: Operation | Sequence[Operation],
+        insertion_point: InsertPoint | None = None,
+    ):
```

Comment:
Actually, I'm relying *a lot* on the return value of `insert`.
Can we make sure to return the inserted `op` (and overload whenever this is a sequence or a single op?).
The use case I have is writing things like:
```
my_op = b.insert(AddOp(lhs, rhs)).res
```
Where you can one-line something quite nicely.